### ISSUE TITLE
[XmlMarshaller] add wrapper for writing XmlElement

### ISSUE THREAD

**Issue by nfx** [doctrine/oxm#13]
Would be good if changing

```
        if ($classMetadata->isCollection($fieldName)) {
            foreach ($fieldValue as $value) {
                $this->doMarshal($value, $writer);
            }
        } else {
            $this->doMarshal($fieldValue, $writer);
        }
```

to

```
                    $mapping = $classMetadata->getFieldMapping($fieldName);
        if ($classMetadata->isCollection($fieldName)) {
            if ($classMetadata->hasFieldWrapping($fieldName)) { 
                $writer->startElement($mapping['wrapper']);
            }            
            foreach ($fieldValue as $value) {
                $this->doMarshal($value, $writer);
            }
            if ($classMetadata->hasFieldWrapping($fieldName)) {
                $writer->endElement();
            }       
        } else {
            $this->doMarshal($fieldValue, $writer);
        }
```

**Comment by richardfullmer** [comment]
@nfx, make PR with tests?

**Comment by richardfullmer** [comment]
Closed by b0c48e64485aa5206fdd3f5b8274cd6ce1debdc6
